### ISSUE TITLE
Cant't receive a refund.

### ISSUE THREAD

**Issue by AnnaMohr83** [UnstoppableSwap/unstoppableswap-gui#200]
I should have been able to receive a refund because the transaction was unsuccessful

I never received a refund.

I can't cancel the transaction because my BTC is locked.

## Steps to Reproduce

1.

2.

3.

4.

## Your Environment

- Node version :
- Operating System and version :

**Comment by binarybaron** [comment]
What state is your swap in? What is displayed in the GUI?

**Comment by AnnaMohr83** [comment]
Nothing is in the GUI and it says they are still waiting for the other
party to lock their monero. My BTC is locke and has been for the past 4
days.

**Comment by AnnaMohr83** [comment]
No, not yet. It still says 0 of 2.

**Comment by AnnaMohr83** [comment]
[image: image.png]

**Comment by binarybaron** [comment]
I can't see the image unfortunately. You need to go to the actual github issue page to attach an image.

The Bitcoin lock transaction hasn't been confirmed yet. This means that the swap hasn't started yet and if it hasn't by now it might be that it won't be able to complete.
You need to wait until the transaction is confirmed and then you need to wait 12 hours and then you can refund your Bitcoin.

**Comment by AnnaMohr83** [comment]
It has been saying "12 hours" for about 3 days now. I just want to refund
it but it hasn't given me the option yet.

**Comment by AnnaMohr83** [comment]
Can I please get an update on this?

**Comment by binarybaron** [comment]
Has the transaction been confirmed?

**Comment by AnnaMohr83** [comment]
It still has zero confirmations and it's been running for many days now.

**Comment by binarybaron** [comment]
I have made a Github issue to introduce a new feature to allow you to recover those funds but it'll take a while for this to be implemented (https://github.com/comit-network/xmr-btc-swap/issues/1664)

For the time being you can do one of two things:
1. Wait for the transaction to be confirmed and as soon as it gets confirmed, you have to wait 12 hours. After those 12 hours have passed you can refund the swap.
2. You can download the [swap command line tool](https://github.com/comit-network/xmr-btc-swap/releases/tag/0.12.3) and execute the `export-bitcoin-wallet` command. You can then import that wallet into [Feather Wallet](https://featherwallet.org/) and then you can send the unconfirmed funds to a Bitcoin address you control. If you want to do this I can help guide you through the process.
3. You can send me the seed phrase of the GUI and I can recover the funds for you (I'll execute the steps above for you on my machine). You have to trust me not to steal funds here. I'm offering this to you. You can decide if you want to do it.

**Comment by AnnaMohr83** [comment]
Which seed phrase for GUI? There is a secret view key, public view key,
secret spend key and private spend key.

**Comment by binarybaron** [comment]
I'm refering to the seed phrase of the UnstoppableSwap GUI, not of the Monero Wallet GUI. It's located in `C:\Users\YOUR_USERNAME\AppData\Roaming\xmr-btc-swap\cli\mainnet\seed.pem` (assuming you are on Windows). You can send it to me via Matrix (`@binarybaron:matrix.org`) or via E-Mail (`help@unstoppableswap.net`). Please also attach a Bitcoin address you want the funds to be sent to.

**Comment by AnnaMohr83** [comment]
Can we try the second option. I can't find the GUI seed anyway...

**Comment by AnnaMohr83** [comment]
Hi. I'm following up. What is the status?

**Comment by binarybaron** [comment]
Hey, assuming the transaction is still unconfirmed please follow the following steps:
1. Download and extract the `swap` tool from https://github.com/comit-network/xmr-btc-swap/releases/tag/0.12.1 to your desktop
2. Open PowerShell
3. Navigate to the desktop: `cd Desktop`
4. Execute the following command: `./swap --data-base-dir C:\Users\YOUR_USERNAME\AppData\Roaming\xmr-btc-swap\cli monero-recovery export-bitcoin-wallet`
5. Download [Sparrow Wallet](https://sparrowwallet.com/)
6. Follow the steps [here](https://github.com/comit-network/xmr-btc-swap/blob/master/docs/asb/README.md#exporting-the-bitcoin-wallet-descriptor) to import the wallet into Sparrow Wallet
7. Go to the transactions tab, find the unconfirmed transaction. Right click -> Cancel Transaction -> Follow the steps

Let me know if you encounter any problems or need any help with the process. I apologise for the inconvenience, we are still trying to figure out the root cause of this.

**Comment by AnnaMohr83** [comment]
I'm going to need some assistance. These directions are not working for me.

**Comment by pscopic** [comment]
> I'm going to need some assistance. These directions are not working for me.

Which step are you stuck on?

**Comment by AnnaMohr83** [comment]
I can't get past steps 3-4.

**Comment by pscopic** [comment]
> I can't get past steps 3-4.

Could you please be more explicit as to why you can't get past steps 3-4 as listed by @binarybaron? What happens when you try?  Could you please include either a screenshot or at minimum a copy paste of what error message is returned when you try to run the swap tool from the powershell command line in step 4?

Precise detail is necessary if you want a useful response. 

> 1. Download and extract the `swap` tool from https://github.com/comit-network/xmr-btc-swap/releases/tag/0.12.1 to your desktop
> 2. Open PowerShell
> 3. Navigate to the desktop: `cd Desktop`
> 4. Execute the following command: `./swap --data-base-dir C:\Users\YOUR_USERNAME\AppData\Roaming\xmr-btc-swap\cli monero-recovery export-bitcoin-wallet`
> 5. Download [Sparrow Wallet](https://sparrowwallet.com/)
> 6. Follow the steps [here](https://github.com/comit-network/xmr-btc-swap/blob/master/docs/asb/README.md#exporting-the-bitcoin-wallet-descriptor) to import the wallet into Sparrow Wallet
> 7. Go to the transactions tab, find the unconfirmed transaction. Right click -> Cancel Transaction -> Follow the steps

**Comment by AnnaMohr83** [comment]
Hi. Have you made any progress with the program to recover my funds? I'm
very anxious to close this matter.

**Comment by binarybaron** [comment]
Has the transaction been confirmed yet? You can write me an email at binarybaron@unstoppableswap.net if you want to have a one-on-one chat via text.

**Comment by AnnaMohr83** [comment]
No, this transaction is still in the Bitcoin Lock phase. I will send an
email to the provided address so we may chat.

**Comment by binarybaron** [comment]
I never received an email from you. Has the issue been resolved?
